rclcpp/src/rclcpp/timer.cpp

```diff
@@ -119,7 +119,9 @@ TimerBase::time_until_trigger()
   int64_t time_until_next_call = 0;
   rcl_ret_t ret = rcl_timer_get_time_until_next_call(
     timer_handle_.get(), &time_until_next_call);
-  if (ret != RCL_RET_OK) {
+  if (ret == RCL_RET_TIMER_CANCELED) {
+    return std::chrono::nanoseconds::max();
```

Comment:
probably we would want to document this? when timer is canceled, it will return that max()? instead, it just appears to be the time for the next trigger?